app/domain/authentication/authn_jwt/restriction_validation/validate_restriction_name.rb

```diff
@@ -0,0 +1,35 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    module RestrictionValidation
+      # Raises error if name of restriction is not according to the nested format
+      #
+      # Valid restriction example:
+      #   - authn-jwt/gitlab/a
+      #   - authn-jwt/gitlab/a/b
+      #   - authn-jwt/gitlab/a/b/c
+      #
+      # Invalid restriction example:
+      #   - authn-jwt/gitlab/a[4]/b
+      #   - authn-jwt/gitlab/a//b
+      #   - authn-jwt/gitlab/a/b[2]
+      #   - authn-jwt/gitlab/a/b/c
+      ValidateRestrictionName = CommandClass.new(
```

Comment:
Why should it be a command class?
It's pure stateless regex validation without dependencies so command class looks like overkill for me

Reply:
i agree. the method here should be implemented in `configuration_jwt_generic_vendor.rb` instead.

Reply:
I think there may be more restriction configuration validators in the future so it should be class with same function name. But i will change it to regular class

Reply:
please let's not do this unless we know for sure how the other classes will be. Let's not write generic code before we know how the instances are similar and how they are different. Let's start with a simple method and generalize in the future if we need to.